File: clojure/grade-school/src/grade_school.clj
(ns grade-school)

;(def db {})

(defn add [db p g] (if (db g) 
                    (assoc db g (into (db g) [p])) 
                    (assoc db g (vector p))))

(defn grade [db g] 
    (into [] (db g)))
  
(defn sorted [db]
  (reduce (fn [x y] (assoc x (first y) (into [] (sort (second y))))) {} (sort db)))

File: javascript/grade-school/grade-school.js
class School {
	
    constructor() {
        this.school = {};
    }

    add(name, grade) {
        if (this.school[grade]) {
            this.school[grade] = this.school[grade].concat(name).sort();
        } else {
            this.school[grade] = [name];
        }
    }

    roster() {
        return this.school;
    }

    grade(y) {
        return Object.keys(this.school).reduce((arr, x) => {
            if (x == y) {
                arr = this.school[x];
            }
            return arr;
        }, []);
    }
}

module.exports = School;